src/main/kotlin/app/revanced/util/resources/ResourceUtils.kt

```diff
@@ -36,22 +50,116 @@ object ResourceUtils {
+     *                                would overwrite an existing file.
      */
-    fun ResourceContext.copyResources(sourceResourceDirectory: String, vararg resources: ResourceGroup) {
+    fun ResourceContext.copyResources(
+        sourceResourceDirectory: String,
+        vararg resources: ResourceGroup,
+        replaceDestinationFiles: Boolean = true
+    ) {
         val classLoader = ResourceUtils.javaClass.classLoader
         val targetResourceDirectory = this["res"]
 
         for (resourceGroup in resources) {
             resourceGroup.resources.forEach { resource ->
+                // Create the target directory if it does not exist
+                this[targetResourceDirectory.resolve(resourceGroup.resourceDirectoryName).absolutePath].mkdir()
                 val resourceFile = "${resourceGroup.resourceDirectoryName}/$resource"
                 Files.copy(
                     classLoader.getResourceAsStream("$sourceResourceDirectory/$resourceFile")!!,
-                    targetResourceDirectory.resolve(resourceFile).toPath(), StandardCopyOption.REPLACE_EXISTING
+                    targetResourceDirectory.resolve(resourceFile).toPath(),
+                    *(if (replaceDestinationFiles) arrayOf(StandardCopyOption.REPLACE_EXISTING) else emptyArray())
                 )
             }
         }
     }
 
+    /**
+     * Copy localized strings from a given directory.
+     */
+    internal fun copyLocalizedStringFiles(context: ResourceContext, directory: String) {
+        val pattern = Pattern.compile("$directory/([-_a-zA-Z0-9]+)/strings\\.xml$")
+
+        var jf: JarFile? = null
+        try {
+            jf = JarFile(this.javaClass.protectionDomain.codeSource.location.toURI().path)
+            val entries = jf.entries()
+            var foundElements = false
+            while (entries.hasMoreElements()) {
+                val match = pattern.matcher(entries.nextElement().name)
+                if (match.find()) {
+                    val languageDirectory = match.group(1)
+                    context.copyResources(
+                        directory,
+                        ResourceGroup(languageDirectory, "strings.xml"),
+                        replaceDestinationFiles = false // Destination files should not exist.
+                    )
+                    foundElements = true
+                }
+            }
+            if (!foundElements) throw PatchException("could not find translated string files")
+        } finally {
+            jf?.close()
+        }
+    }
+
+    /**
+     * Used by Gradle task to merge the English patch strings into a single file for translations.
+     */
+    @JvmStatic
+    fun main(args: Array<String>) {
+        if (args.isEmpty() || args.size % 2 != 0) throw IllegalArgumentException()
+
+        for (i in args.indices step 2) {
+            mergeXMLFiles(args[i], args[i + 1])
+        }
+    }
```

Comment:
I have yet to understand the reason for this entirely.
The XML files are in the resources folder, which Crowdin can pick up, is there any reason why a Gradle task is necessary